tools/archives/pyvo_integration/astronomical_archives.xml

```diff
@@ -0,0 +1,977 @@
+                <param name="query_type" value="obscore_query" />
+                <param name="dataproduct_type" value="image" />
+                <param name="obs_title" value="190807041_AP_B001" />
+            </conditional>
+            <output name="output" count="1">
+              <assert_contents>
+                  <has_text_matching expression="SIMPLE  =" />
+              </assert_contents>
+            </output>
+        </test>
+        <test expect_num_outputs="1">
+            <param name="output_selection" value="c"/>
+            <param name="number_of_files" value="1"/>
+            <conditional name="archive_selection">
+                <param name="archive_type" value="registry"/>
+                <param name="keyword" value="apertif"/>
+            </conditional>
+            <conditional name="query_selection">
+                <param name="query_type" value="obscore_query" />
+                <param name="dataproduct_type" value="image" />
+                <param name="obs_title" value="190807041_AP_B001"/>
+            </conditional>
+            <output name="output_csv" count="1">
+              <assert_contents>
+                  <has_line line="https://vo.astron.nl/getproduct/APERTIF_DR1/190807041_AP_B001/image_mf_02.fits," />
+              </assert_contents>
+            </output>
+        </test>
+        <test expect_num_outputs="1">
+            <param name="output_selection" value="c"/>
+            <param name="number_of_files" value="1"/>
+            <conditional name="archive_selection">
+                <param name="archive_type" value="registry"/>
+                <param name="keyword" value="apertif"/>
+            </conditional>
+            <conditional name="query_selection">
+                <param name="query_type" value="obscore_query" />
+                <param name="dataproduct_type" value="image" />
+                <param name="obs_title" value="190807042_AP_B006"/>
+            </conditional>
+            <output name="output_csv" count="1">
+              <assert_contents>
+                  <has_line line="https://vo.astron.nl/getproduct/APERTIF_DR1/190807042_AP_B006/image_mf_04.fits," />
+              </assert_contents>
+            </output>
+        </test>
+        <test expect_num_outputs="1">
+          <param name="output_selection" value="c"/>
+          <param name="number_of_files" value="1"/>
+          <conditional name="archive_selection">
+              <param name="archive_type" value="registry"/>
+              <param name="keyword" value="simbad"/>
+          </conditional>
+          <conditional name="query_selection">
+              <param name="query_type" value="obscore_query" />
+          </conditional>
+          <output name="output_error" count="1">
+            <assert_contents>
+                <has_text text="No obscore table in the archive"/>
+            </assert_contents>
+          </output>
+        </test>
+        <test expect_num_outputs="1">
```

Comment:
Mh, the tool does not fail when we have an error in the query?

Would that not be the expected behaviour from a users point of view.
You can test for error code and such. But this needs to be implemented in your python script I guess, that you really fail.

Reply:
when a query fail we update the summary output to inform the user so he can possibly fix the error in the query, if the tool failed the user would not get the summary output right or am I mistaken?

Reply:
Yes, if it fails the outputs is not returned. But you get the stdout/stderr.

Reply:
ok I thought it would be more user friendly if we informed the user through some summary file but we could just output everything in the stdout if you think it would be better no problem

Reply:
Your call, I don't know :)
I'm happy to +1 it as it is :)